src/lib/libemval.js

```diff
@@ -122,50 +122,46 @@ var LibraryEmVal = {
   _emval_new_u16string__deps: ['$Emval'],
   _emval_new_u16string: (v) => Emval.toHandle(UTF16ToString(v)),
 
+  $emGlobalThis__internal: true,
 #if SUPPORTS_GLOBALTHIS
-  $emval_get_global: () => globalThis,
-#elif !DYNAMIC_EXECUTION
-  $emval_get_global: () => {
-    if (typeof globalThis == 'object') {
+  $emGlobalThis: 'globalThis',
+#else
+  $getGlobalThis__internal: true,
+  $getGlobalThis: () => {
+    if (typeof globalThis != 'undefined') {
       return globalThis;
     }
+#if DYNAMIC_EXECUTION
+    return new Function('return this')();
+#else
     function testGlobal(obj) {
-      obj['$$$embind_global$$$'] = obj;
-      var success = typeof $$$embind_global$$$ == 'object' && obj['$$$embind_global$$$'] == obj;
-      if (!success) {
-        delete obj['$$$embind_global$$$'];
-      }
+      // Use __emGlobalThis as a test symbol to see if `obj` is indeed the
+      // global object.
+      obj['__emGlobalThis'] = obj;
+      var success = typeof __emGlobalThis == 'object' && obj['__emGlobalThis'] === obj;
+      delete obj['__emGlobalThis'];
       return success;
     }
-    if (typeof $$$embind_global$$$ == 'object') {
-      return $$$embind_global$$$;
+    if (typeof global != 'undefined' && testGlobal(global)) {
```

Comment:
The change from `=== 'object'` to `!= 'undefined'` may run into problems if `global = 42`? Writing a field to a Number traps.

I guess this trapped before on `null`, though...